### ISSUE TITLE
RegExp issues with Style/VariableNumber in snake_case style

### ISSUE THREAD

**Issue by tdeo** [bbatsov/rubocop#3596]
When using the `Style/VariableNumber` with `EnforcedStyle: snake_case`, the regexp defined https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/cop/mixin/configurable_numbering.rb#L10 doesn't allow any use of number in the middle of a variable name.

--------

## Expected behavior

With this input file
```shell
$ cat demo.rb
ec2_user = 'blublu'
ec_2_user = 'blublu'
ec_2user = 'blublu'
```

The `Style/VariableNumber` cop should accept at least one of this options, if not all.

## Actual behavior

When running rubocop (having `EnforcedStyle: snake_case` for `Style/VariableNumber` cop, I get:
```shell
$ ruby -S rubocop -c ../devtools/rubocop/rubocop.yml demo.rb
Inspecting 1 file
W

Offenses:

demo.rb:1:1: C: Use snake_case for variable numbers.
ec2_user = 'blublu'
^^^^^^^^
demo.rb:2:1: C: Use snake_case for variable numbers.
ec_2_user = 'blublu'
^^^^^^^^^
demo.rb:3:1: C: Use snake_case for variable numbers.
ec_2user = 'blublu'
^^^^^^^^

1 file inspected, 3 offenses detected
```

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ ruby -S rubocop -V
0.43.0 (using Parser 2.3.1.2, running on ruby 2.2.4 x86_64-darwin15)
```

**Comment by bbatsov** [comment]
Try version 0.44.1.

**Comment by tdeo** [comment]
All of them work with 0.44.1

Thanks.
